### ISSUE TITLE
WebXR improvements

### ISSUE THREAD

**Issue by deltakosh** [BabylonJS/Babylon.js#6958]
List of features we want to add as soon as the spec will be ready:

- [ ] Blocked on spec/browser implementation
  - [x] Mirroring to canvas or custom canvas (WebXRExperienceHelper's enterXRAsync accepts a WebXRManagedOutputCanvas, currently this must be created from the active babylon canvas and does not mirror, waiting on browser implementation to support this)
  - [x] AR scenario support (the current spec only has VR parity)
  - [x] Additional controller/input support, the current spec exposes the browser gamepad corresponding to the controller, supporting future controllers, head gaze, select event, maybe will work differently in the future
  - [ ] Multiview support, currently the XR layer exposed does not support multiview, due to the work done to support multiview for webVR, supporting this once the spec is finalized should be somewhat straitforward by using camera.outputRenderTarget and MultiviewRenderTarget class

  - [x] Hand-tracking (based on this: https://github.com/immersive-web/webxr-hand-input)

**Comment by RaananW** [comment]
According to https://blog.mozvr.com/multiview-on-webxr/ , WebXR theoretically supports multiview, if antialias is set to false (See "Waht's next" at the end).

**Comment by RaananW** [comment]
Mirroring - See note here - https://www.w3.org/TR/webxr/#xrrenderstate-interface :
""Note: At this point the XRRenderState will only have an output canvas if it has composition disabled, but future versions of the specification are likely to introduce methods for setting output canvas' that support more advanced uses like mirroring and layer compositing that will require composition.""

**Comment by RaananW** [comment]
Mirroring - we should aos enable spectator mode if possible (for example: https://github.com/immersive-web/webxr-samples/blob/master/spectator-mode.html)

**Comment by RaananW** [comment]
AR is supported, not including AR-specific features such as hit test and plane detection. A new ticket will be created for these features

**Comment by deltakosh** [comment]
Thanks!

**Comment by deltakosh** [comment]
I would recommend to close this and create new entries for the remaining pieces
